src/overlays/actors/ovl_En_Aob_01/z_en_aob_01.c

```diff
@@ -488,124 +554,152 @@ void func_809C1EC8(EnAob01* this, PlayState* play) {
             if (this->collider.base.ocFlags2 & OC2_HIT_PLAYER) {
                 this->actor.flags |= ACTOR_FLAG_10000;
                 func_800B8614(&this->actor, play, 100.0f);
-                this->unk_2D2 |= 8;
-                this->actionFunc = func_809C21E0;
+                this->stateFlags |= ENAOB01_FLAG_TALKING_TO_PLAYER_HOLDING_DOG;
+                this->actionFunc = EnAob01_BeforeRace_Talk;
             }
-        } else if (Actor_ProcessTalkRequest(&this->actor, &play->state) && (this->unk_2D2 & 0x100)) {
-            this->unk_2D2 &= ~0x100;
-            this->unk_2E6 = this->trackTarget;
-            this->unk_2EC = this->headRot;
-            this->unk_2F2 = this->torsoRot;
-            func_809C16DC(this, play);
-            this->actionFunc = func_809C21E0;
+        } else if (Actor_ProcessTalkRequest(&this->actor, &play->state) &&
+                   (this->stateFlags & ENAOB01_FLAG_PLAYER_CAN_TALK)) {
+            this->stateFlags &= ~ENAOB01_FLAG_PLAYER_CAN_TALK;
+            this->prevTrackTarget = this->trackTarget;
+            this->prevHeadRot = this->headRot;
+            this->prevTorsoRot = this->torsoRot;
+            EnAob01_BeforeRace_HandleConversation(this, play);
+            this->actionFunc = EnAob01_BeforeRace_Talk;
         } else {
-            this->unk_2D2 &= ~0x100;
+            this->stateFlags &= ~ENAOB01_FLAG_PLAYER_CAN_TALK;
             if ((this->actor.xzDistToPlayer < 100.0f) && !(this->collider.base.ocFlags2 & OC2_HIT_PLAYER)) {
-                this->unk_2D2 |= 0x100;
+                this->stateFlags |= ENAOB01_FLAG_PLAYER_CAN_TALK;
                 func_800B8614(&this->actor, play, 100.0f);
             }
         }
     }
 }
 
-void func_809C21E0(EnAob01* this, PlayState* play) {
+/**
+ * Handles the actual action of talking with the player before the race begins.
+ */
+void EnAob01_BeforeRace_Talk(EnAob01* this, PlayState* play) {
     u8 talkState = Message_GetState(&play->msgCtx);
 
     Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 4, 4000, 1);
 
-    if ((this->unk_2D2 & 8) && !func_809C2EC4(this, play)) {
-        if ((this->unk_210 != 0) && (this->unk_210 != 0x3535) && (this->unk_210 != 0x3524) &&
-            (this->unk_210 != 0x3548) && (this->unk_210 != 0x3549) && (this->unk_210 != 0x354A)) {
-            this->unk_210 = 0x3523;
+    //! @bug: This block of code acts as a failsafe for when the player triggered this conversation by
+    //! bumping into the racetrack owner while holding the dog and, at the same time, threw or dropped
+    //! the dog frame-perfectly by pressing A. In this case, the code successfully returns the racetrack
+    //! owner to her idle state. Unfortunately, there is another way to let go of the dog besides pressing
+    //! A to drop or throw it; the player can also drop the dog by shielding. If this shield drop is done
```

Comment:
```suggestion
    //! owner to her idle state. Unfortunately, the player can also drop the dog by shielding. If this shield drop is done
```
I think part of this is fluff, and this still gets the point across